tests/container_tests/test_zip_container.py

```diff
@@ -559,3 +560,258 @@ def func():
+                       NO_DIRECTORY_FILENAME_LIST["testfile1_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["dir2_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile2_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir3_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile3_name"]),
+          NO_DIRECTORY_FILENAME_LIST["testfile4_name"]],
+         True],
+        # not normalized path beyond root
+        ['//..//',
+         [os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile1_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["dir2_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile2_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir3_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile3_name"]),
+          NO_DIRECTORY_FILENAME_LIST["testfile4_name"]],
+         True],
+        # not normalized path beyond root
+        ['testdir2//..//',
+         [os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile1_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["dir2_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile2_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir3_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile3_name"]),
+          NO_DIRECTORY_FILENAME_LIST["testfile4_name"]],
+         True],
+        # starting with slash
+        ['/',
+         [os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile1_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                       NO_DIRECTORY_FILENAME_LIST["dir2_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile2_name"]),
+          os.path.join(NO_DIRECTORY_FILENAME_LIST["dir3_name"],
+                       NO_DIRECTORY_FILENAME_LIST["testfile3_name"]),
+          NO_DIRECTORY_FILENAME_LIST["testfile4_name"]],
+         True]
+    ])
+    def test_list(self, path_or_prefix, expected_list, recursive):
+        with self.fs_handler.open_as_container(self.zip_file_name) as handler:
+            zip_generator = handler.list(path_or_prefix,
+                                         recursive=recursive)
+            zip_list = list(zip_generator)
+            self.assertEqual(sorted(expected_list),
+                             sorted(zip_list))
+
+    @parameterized.expand([
+        # non_exist_file
+        ['does_not_exist', FileNotFoundError],
+        # not exist but share the prefix
+        ['t', FileNotFoundError],
+        # broken path
+        ['testdir1//t/', FileNotFoundError],
```

Comment:
How about using `NO_DIRECTORY_FILENAME_LIST["dir1_name"]` insted of `testdir1` ?